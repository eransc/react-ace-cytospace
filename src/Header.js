import React from "react";
import { useDispatch } from "react-redux";
import { addNode, addEdge } from "./store/actions";

var divHeader = {
  textAlign: "center",
};

const Header = () => {
  const dispatch = useDispatch();

  const addNodeItem = () => {
    const node = {
      data: { id: "6", label: "Node 6" },
      position: { x: 400, y: 100 },
    };

    dispatch(addNode(node));
  };

  const addEdgeItem = () => {
    const edge = {
      data: {
        source: "6",
        target: "two",
        label: "Edge from Node1 to Node2",
      }
    };

    dispatch(addEdge(edge));
  };

  return (
    <>
      <input type="button" value="Add Node" onClick={addNodeItem} />
      <input type="button" value="Add Edge" onClick={addEdgeItem} />
      <h1 style={divHeader}>Graph Editor</h1>
      <hr />
    </>
  );
};

export default Header;
