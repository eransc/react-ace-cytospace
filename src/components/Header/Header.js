import React from "react";
import { useDispatch } from "react-redux";
import { addNode, addEdge } from "../../store/actions";

var divHeader = {
  textAlign: "center",
};

const Header = ({text}) => {
  // const dispatch = useDispatch();

  // const addNodeItem = () => {
  //   const node = {
  //     data: { id: "6", label: "Node 6" },
  //     position: { x: 400, y: 100 },
  //   };

  //   dispatch(addNode(node));
  // };

  // const addEdgeItem = () => {
  //   const edge = {
  //     data: {
  //       source: "6",
  //       target: "two",
  //       label: "Edge from Node1 to Node2",
  //     }
  //   };

  //   dispatch(addEdge(edge));
  // };

  return (
    <>
      {/* <input type="button" value="Add Node" onClick={addNodeItem} />
      <input type="button" value="Add Edge" onClick={addEdgeItem} /> */}
      <h1 data-testid="h1tag" className="fancy" style={divHeader}>{text}</h1>
      <hr />
    </>
  );
};

export default Header;
