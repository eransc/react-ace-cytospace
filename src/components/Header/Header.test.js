import React from "react";
import { render, cleanup } from "@testing-library/react";

import Header from "./Header";

afterEach(cleanup);

it("insert text into h1", () => {
  const { getByTestId, getByText } = render(<Header text="Hello" />);
  expect(getByTestId("h1tag")).toHaveTextContent("Hello");
  expect(getByText("Hello")).toHaveClass("fancy");
});
