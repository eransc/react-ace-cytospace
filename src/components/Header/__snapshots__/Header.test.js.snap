// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <h1
    class="fancy"
    data-testid="h1tag"
    style="text-align: center;"
  >
    Hello2
  </h1>
  <hr />
</DocumentFragment>
`;
